src/client/components/app-header/app-header.vue

```diff
@@ -1,61 +1,216 @@
+      <menu-icon :isOpen="menuIsOpen" />
+    </button>
   </header>
 </template>
 
 <script>
-import menuI18n from '../../static/data/menu.json'
 import LanguageSelector from '../language-selector'
+import SmartLink from '../smart-link'
+import MenuIcon from '../menu-icon'
 
 export default {
-  components: { LanguageSelector },
-  props: ['contentId'],
+  components: { LanguageSelector, SmartLink, MenuIcon },
+  props: ['contentId', 'menuI18n'],
   data () {
-    return { menuI18n }
+    return {
+      menuIsOpen: false,
+    }
   },
   computed: {
     locale() { return this.$i18n.locale },
     menu() { return this.menuI18n[this.locale] },
+    slicedMenu() { return (this.menu.callToAction || this.$i18n.locales) ? this.menu.items.slice(0, 3) : this.menu.items.slice(0, 5)}
   },
+  methods: {
+    toggleMenu () {
+      this.menuIsOpen = !this.menuIsOpen
+    },
+  }
 }
 </script>
 
 <style>
 @import '../app-core/index.css';
 
-.app-header__menu-link {
+.app-header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  position: sticky;
+  top: 0;
+  left: 0;
+  z-index: 1;
+  padding: 0 var(--spacing-default);
+  width: 100%;
+  height: var(--app-header-mobile-height);
+  background-color: var(--background-color);
+  box-shadow: 0 2px 15px 0 rgba(214,214,214,.5);
+}
+
+.app-header__navigation {
+  position: absolute;
+  top: var(--app-header-mobile-height);
+  left: 0;
+  display: none;
+  width: 100%;
+  background-color: var(--background-color);
+  box-shadow: 0px 8px 10px rgba(214,214,214,.3);
+  transition: all .2s ease-in-out;
+}
+
+.app-header__navigation--open {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding-bottom: var(--spacing-default);
+}
+
+.app-header__navigation-list {
+  margin: 0 0 var(--spacing-default) 0;
+  padding: 0;
+  text-align: center;
+}
+
+.app-header__navigation-item {
+  height: auto;
+  margin-bottom: var(--spacing-double);
+  list-style-type: none;
+}
+
+.app-header__navigation-link {
+  padding: var(--spacing-default) 0;
+  color: var(--text-color);
+  text-transform: uppercase;
+  text-decoration: none;
+}
+
+.app-header__navigation-link:focus,
+.app-header__navigation-link:hover {
+  cursor: pointer;
+  outline: none;
+  border-bottom: 3px solid var(--action-color);
+}
+
+.app-header__identity {
+  display: flex;
+  align-items: center;
+  color: var(--text-color);
   text-decoration: none;
 }
 
-.app-header__menu-link {
-  border-bottom-width: 2px;
-  border-bottom-style: solid;
-  border-bottom-color: transparent;
+.app-header__identity:hover,
+.app-header__identity:focus {
+  border: none;
+}
+
+.app-header__logo {
+  margin-right: .3rem;
+  height: 40px;
 }
 
-.app-header__menu-link:focus,
-.app-header__menu-link:hover {
-  color: var(--action-color);
+.app-header__title {
+  display: none;
+  font-size: var(--font-size-default);
+  font-weight: lighter;
+}
+
+.app-header__call-to-action {
```

Comment:
`.app-header__button`, if we would have more things that look like this but doing different things, we would both call them `__button`.